File: rust/src/java/test_object.rs
use javawithrust::prelude::*;


#[jclass(net.totobirdcreations.robot.rust.TestObject)]
pub struct TestObject {
    content : String
}
impl TryFrom<TestObject> for InvocationArg {
    type Error = errors::J4RsError;
    fn try_from(arg : TestObject) -> errors::Result<InvocationArg> {
        return Ok(InvocationArg::new(&arg, __jwrs_classname_TestObject!()));
    }
}


#[jfuncs]
impl TestObject {
    fn create(text : String) -> /*Result<*/TestObject/*, String>*/ {
        return Ok(TestObject {
            content : text
        });
    }
    fn concat(mut this : TestObject, text : String) -> /*Result<*/TestObject/*, String>*/ {
        this.content += &text;
        return Ok(this);
    }
}

File: java/src/main/java/net/totobirdcreations/robot/rust/TestObject.java
package net.totobirdcreations.robot.rust;

import org.astonbitecode.j4rs.api.Instance;
import org.astonbitecode.j4rs.api.java2rust.Java2RustUtils;


public class TestObject {

    public String content;


    public TestObject(String start) {
        this.content = start;
    }


    private static native Instance<TestObject> create(Instance<String> text);

    public static TestObject createFromRust(String next) {
        return Java2RustUtils.getObjectCasted(TestObject.create(
            Java2RustUtils.createInstance(next)
        ));
    }

}
